synapse/handlers/room_summary.py

```diff
@@ -1096,9 +1096,48 @@ async def get_room_summary(
 
                 room_summary["membership"] = membership or "leave"
         else:
-            # TODO federation API, descoped from initial unstable implementation
-            #      as MSC needs more maturing on that side.
-            raise SynapseError(400, "Federation is not currently supported.")
+            # Reuse the hierarchy query over federation
+            if remote_room_hosts is None:
+                raise SynapseError(400, "Missing via to query remote room")
+
+            fed_rooms = await self._summarize_remote_room(
+                _RoomQueueEntry(room_id, remote_room_hosts),
+                suggested_only=True,
+                max_children=1,
+                exclude_rooms=[],
+            )
+
+            for room_entry in fed_rooms:
+                room = room_entry.room
+                fed_room_id = room_entry.room_id
+
+                if fed_room_id == room_id:
+                    # If no user is specified, test against an invalid userid
+                    if requester is None:
+                        requester = ""
```

Comment:
This is hacky; perhaps we could update `_is_remote_room_accessible` to handle `requester=None` to mean an unauthenticated local user?

For local rooms, we already do this with `_is_local_room_accessible` indirectly.

Reply:
Dammit, you noticed me being lazy. Yes, I will fix that! :D